test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_cad.cxx

```diff
@@ -889,3 +945,182 @@ INSTANTIATE_TEST_SUITE_P (t8_gtest_check_2d_element_curved_cad_surface, class_2d
+  }
+  t8_cmesh_destroy (&cmesh);
+
+#else  /* !T8_WITH_OCC */
+  SC_ABORTF ("OCC not linked");
+#endif /* T8_WITH_OCC */
+}
+
+#if T8_WITH_OCC
+TEST (t8_gtest_geometry_cad_pyramid, linked_faces)
+{
+  /* clang-format off */
+  /* Reference coordinates of each face vertex and the centroid of the pyramid */
+  double test_ref_coords[75]
+    = { 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 0.6, 0.6, 0.2, 2.0, 2.0, 2.0,   // face 0
+        1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0, 1.0, 0.6, 0.6, 0.2, 2.0, 2.0, 2.0,   // face 1
+        0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 1.0, 0.6, 0.6, 0.2, 2.0, 2.0, 2.0,   // face 2
+        1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 1.0, 0.6, 0.6, 0.2, 2.0, 2.0, 2.0,   // face 3
+        0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.6, 0.6, 0.2 }; // face 4
+    /* The 2.0's at face 0-3 are placeholders, because these faces only have 3 vertices. */
+
+  double surface_test_return_coords_quad[15]
+    = { 0.0, 0.0, 0.0,    // face vertex 0
+        1.0, 0.0, 0.0,    // face vertex 1
+        0.0, 1.0, 0.0,    // face vertex 2
+        1.0, 1.0, 0.0,    // face vertex 3
+        0.6357349585, 0.6357349585, 0.1438333846 };  // shifted centroid of the pyramid
+  double surface_test_return_coords_tri[12]
+    = { 1.0, 0.0, 0.0,    // face vertex 0
+        1.0, 1.0, 0.0,    // face vertex 1
+        1.0, 1.0, 1.0,    // face vertex 2
+        0.6737526363, 0.5930049972, 0.1275036282 };  // shifted centroid of the pyramid
+
+  double surface_parameters[40]
+    = { 0, 1, 0, 0, 1, 1, 2, 2,   // face 0
+        0, 0, 0, 1, 1, 1, 2, 2,   // face 1
+        0, 0, 0, 1, 1, 1, 2, 2,   // face 2
+        0, 1, 0, 0, 1, 1, 2, 2,   // face 3
+        0, 0, 0, 1, 1, 0, 1, 1 }; // face 4
+    /* The 2's of face 0-3 are placeholders, because these faces only have 3 vertices. */
+  /* clang-format on */
+
+  for (int i_faces = 0; i_faces < 5; i_faces++) {
+    t8_test_geometry_cad_pyramid (i_faces, -1, surface_parameters + i_faces * 8, test_ref_coords,
+                                  (i_faces < 4 ? surface_test_return_coords_tri : surface_test_return_coords_quad));
+  }
+}
+
+TEST (t8_gtest_geometry_cad_pyramid, linked_edges)
+{
+  /* clang-format off */
+  double test_ref_coords[72]
+    = { 0.0, 1.0, 0.0, 0.0, 0.5, 0.0, 0.0, 0.0, 0.0,   // edge 0
+        1.0, 0.0, 0.0, 1.0, 0.5, 0.0, 1.0, 1.0, 0.0,   // edge 1
+        0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0,   // edge 2
+        1.0, 1.0, 0.0, 0.5, 1.0, 0.0, 0.0, 1.0, 0.0,   // edge 3
+        1.0, 1.0, 1.0, 0.5, 0.5, 0.5, 0.0, 0.0, 0.0,   // edge 4
+        1.0, 1.0, 1.0, 1.0, 0.5, 0.5, 1.0, 0.0, 0.0,   // edge 5
+        1.0, 1.0, 1.0, 1.0, 1.0, 0.5, 1.0, 1.0, 0.0,   // edge 6
+        1.0, 1.0, 1.0, 0.5, 1.0, 0.5, 0.0, 1.0, 0.0 }; // edge 7
+  double curve_test_return_coords[9]
+    = { 0.0, 0.0, 0.0,    // edge vertex 0
+        0.4999500215, 0.0000523914, 0.4000007585,    // center of edge
```

Comment:
Maybe `T8_PRECISION_SQRT_EPS` suffices and we can use 0.5 0 0.4

Reply:
Nope it doesn't